### PR TITLE
Feature/ulid

### PR DESCRIPTION
**Pull request by mambax7**

### CODE REVIEW

**Review by geekwright — CHANGES_REQUESTED** [review]
This is neat!

Can you add a couple of tests?
- generate 2 ULIDs and compare to assert they are unequal
- generate 2 ULIDs and sort to verify they collate properly

Looks good!

**Comment by mambax7** [comment]
OK, done!

**Comment by mambax7** [comment]
Richard, did you get a chance to look into it?

File: src/Ulid.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ */
+class Ulid
+{
+    const ENCODING_CHARS = '0123456789ABCDEFGHJKMNPQRSTVWXYZ';
+    const ENCODING_LENGTH = 32;
+    /**
+     * Last generated timestamp.
+     * @var int
+     */
+    private static $lastTime = 0;
```
Comment (geekwright) [inline src/Ulid.php:31]: This change adding the static needs to be reverted. This should be stateless.

The testing would be generating one ULID, followed by a usleep() then generating another ULID  to compare for proper coalation. Best  guess would be usleep(50000)?

**Review by geekwright — CHANGES_REQUESTED** [review]

**Comment by mambax7** [comment]
Please see the latest changes

**Comment by mambax7** [comment]
FYI - here the tests fail on PhpUnit 7.1, and 7.4, but on my PHPUnit 9.6.13 they passed

**Review by ggoffy — APPROVED** [review]

**Review by ggoffy — APPROVED** [review]

**Review by ggoffy — APPROVED** [review]

**Review by ggoffy — APPROVED** [review]

**Review by ggoffy — APPROVED** [review]

**Review by ggoffy — APPROVED** [review]

**Review by ggoffy — APPROVED** [review]

**Review by geekwright — APPROVED** [review]

**Pull request merged**